pkg/server/chaosd/jvm.go

```diff
@@ -0,0 +1,214 @@
+// Copyright 2020 Chaos Mesh Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package chaosd
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"text/template"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/log"
+	"go.uber.org/zap"
+
+	"github.com/chaos-mesh/chaosd/pkg/core"
+)
+
+const ruleTemplate = `
+RULE {{.Name}}
+CLASS {{.Class}}
+METHOD {{.Method}}
+AT ENTRY
+IF true
+DO 
+	{{.Do}};
+ENDRULE
+`
+
+const stressRuleTemplate = `
+RULE {{.Name}}
+STRESS {{.StressType}}
+{{.StressValueName}} {{.StressValue}}
+ENDRULE
+`
+
+const gcRuleTemplate = `
+RULE {{.Name}}
+GC
+ENDRULE
+`
+
+type jvmAttack struct{}
+
+var JVMAttack AttackType = jvmAttack{}
+
+const bmInstallCommand = "bminstall.sh -b -Dorg.jboss.byteman.transform.all -Dorg.jboss.byteman.verbose -p %d %d"
+const bmSubmitCommand = "bmsubmit.sh -p %d -%s %s"
+
+func (j jvmAttack) Attack(options core.AttackConfig, env Environment) (err error) {
+	attack := options.(*core.JVMCommand)
+
+	if attack.Type == core.JVMAttachType {
+		return j.attach(attack)
+	} else if attack.Type == core.JVMInstallRuleType {
+		return j.installRule(attack)
+	}
+
+	return errors.Errorf("attack type %s not supported", attack.Type)
+}
+
+func (j jvmAttack) attach(attack *core.JVMCommand) error {
+	var err error
+
+	bmInstallCmd := fmt.Sprintf(bmInstallCommand, attack.Port, attack.Pid)
```

Comment:
func attach here use install bash command , but func install rule use submit bash command. I think the naming here is not proper.

Reply:
rename it in https://github.com/chaos-mesh/chaosd/pull/33/commits/49f24410a36dcbfbc842ace7b2b52ba5e530e2ad

Reply:
rename it in https://github.com/chaos-mesh/chaosd/pull/33/commits/49f24410a36dcbfbc842ace7b2b52ba5e530e2ad

Reply:
rename it in https://github.com/chaos-mesh/chaosd/pull/33/commits/49f24410a36dcbfbc842ace7b2b52ba5e530e2ad